security/oc_tls.c

```diff
@@ -2112,6 +2110,54 @@ write_application_data(oc_tls_peer_t *peer)
   }
 }
 
+static void
+oc_tls_handshake(oc_tls_peer_t *peer)
+{
+  int ret = mbedtls_ssl_handshake(&peer->ssl_ctx);
+  if (ret < 0 && ret != MBEDTLS_ERR_SSL_WANT_READ &&
+      ret != MBEDTLS_ERR_SSL_WANT_WRITE) {
+#ifdef OC_DEBUG
+    char buf[256];
+    mbedtls_strerror(ret, buf, 256);
+    OC_ERR("oc_tls: mbedtls_error: %s", buf);
+#endif /* OC_DEBUG */
+    oc_tls_free_peer(peer, false);
+    return;
+  }
+  if (ret == 0) {
+    oc_tls_handler_schedule_write(peer);
+    return;
+  }
+}
+
+#ifdef OC_TCP
+#ifdef OC_TCP_ASYNC_CONNECT
+static void
+oc_tls_on_tcp_connect(const oc_endpoint_t *endpoint, int state, void *user_data)
+{
+  (void)user_data;
+  OC_DBG("oc_tls_on_tcp_connect: %d", state);
+  if (state == OC_TCP_SOCKET_STATE_CONNECTED) {
+    oc_tls_peer_t *peer = oc_tls_get_peer(endpoint);
+    if (peer == NULL) {
+      OC_ERR("oc_tls_connect_tcp_session: peer not found for endpoint");
+      return;
+    }
+    oc_tls_handshake(peer);
+  }
+}
+#endif /* OC_TCP_ASYNC_CONNECT */
+
+void
+oc_tls_connect_tcp_session(oc_tcp_on_connect_event_t *event)
+{
+  if (event->fn != NULL) {
+    event->fn(&event->endpoint, event->state, event->fn_data);
+  }
+  oc_tcp_on_connect_event_free(event);
+}
+#endif /* OC_TCP */
```

Comment:
This function belongs to `/api/oc_tcp.c`- eg SECURITY== 0 and rename it to `oc_tcp_on_connect`